tests/ffi/tests.h

```diff
@@ -116,6 +116,15 @@ const rust::Vec<uint8_t> &c_return_ref_rust_vec(const C &c);
 rust::Vec<uint8_t> &c_return_mut_rust_vec(C &c);
 rust::Vec<rust::String> c_return_rust_vec_string();
 rust::Vec<bool> c_return_rust_vec_bool();
+rust::Option<rust::Box<Shared>> c_return_rust_option_box();
+rust::Option<const Shared*> c_return_rust_ref_option_shared();
+rust::Option<Shared*> c_return_rust_mut_option_shared();
+rust::Option<Shared*> c_return_rust_pin_mut_option_shared();
+rust::Option<const C*> c_return_rust_ref_option_opaque();
+rust::Option<C*> c_return_rust_pin_mut_option_opaque();
+rust::Option<const uint8_t*> c_return_rust_ref_option_native();
+rust::Option<uint8_t*> c_return_rust_mut_option_native();
+rust::Option<uint8_t*> c_return_rust_pin_mut_option_native();
```

Comment:
I think that didn't work but I can try again

Reply:
@kuecks to do this, you'll need a specialization for `rust::Option<T&>` that uses a raw pointer as the field in storage, but otherwise exposes `T&` APIs.

Reply:
Hrmm rather than that, maybe it would make more sense to introduce a new type, `rust::OptionRef<T>`. Potentially `rust::Option<T>` would also be renamed to `rust::OptionBox<T>` as well, or maybe I can restrict `rust::Option<T>` to only exist when T == Box<U>

Reply:
`rust::OptionRef<T>` seems reasonable as well. There's prior art for that in other codebases:

- Chromium https://chromium.googlesource.com/chromium/src/base/+/refs/heads/main/types/optional_ref.h
- TypeSafe https://github.com/foonathan/type_safe/blob/main/include/type_safe/optional_ref.hpp

Some more discussion here https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2020/p1683r0.html